ServerCore/Pages/EventSpecific/FAQ.cshtml

```diff
@@ -5,6 +5,46 @@
     ViewData["AdminRoute"] = "/EventSpecific/FAQ";
     ViewData["AuthorRoute"] = "/EventSpecific/FAQ";
     ViewData["PlayRoute"] = "/EventSpecific/FAQ";
+    var eventPage = Model as ServerCore.ModelBases.EventSpecificPageModel;
+    var Event = eventPage?.Event;
 }
 
-<partial name="../Resources/@(Model.Event.HomePartial ?? "Default")/FAQPartial.cshtml" />
\ No newline at end of file
+@if (Event.EventBegin.Year < 2024)
+{
+    <partial name="../Resources/@(Event.HomePartial ?? "Default")/FAQPartial.cshtml" />
+}
+else
+{
+    if (!string.IsNullOrEmpty(Event.FAQContent))
+    {
+        @Html.Raw(Event.FAQContent);
+    }
+    <style>
+        #faq-frame.unsized {
+            width: 100%;
+            height: 100vh;
+        }
+    </style>
+    <script type="text/javascript">
+        window.addEventListener("load", () => {
+            const iframe = document.querySelector("#faq-frame");
+            if ((iframe !== null) && (iframe !== undefined)) {
+                iframe.contentWindow.postMessage("", "*");
+            }
+        });
+        window.addEventListener("message", (e) => {
+            if ((e.data.width > 0) && (e.data.height > 0)) {
+                const iframe = document.querySelector("#faq-frame");
+                iframe.classList.remove("unsized");
+                iframe.width = e.data.width + 30;
+                iframe.height = e.data.height + 30;
+            }
+        });
+    </script>
+    <environment include="Development">
```

Comment:
Suggest revising this so that it gets the path from FileManager.GetPuzzleDirectoryAsync (which would of course need to become public). This would make it more stable in the case that, say, the local storage path structure changes in the dev environment (since we don't really control that prefix).

Reply:
Done for all hard-coded urls.